site/profile/manifests/ncsa/allow_ssh_from_login.pp

```diff
@@ -0,0 +1,45 @@
+# Allow incoming ssh from any login nodes
+#
+# @summary Allow incoming ssh from any login nodes
+#
+# @param
+#   login_nodelist
+#   Type: Array
+#   Desc: one or more hostnames / IPs / CIDRs
+#         Note: must contain at least 1 item
+# @param
+#   allow_groups
+#   Type: Array
+#   Desc: one or more LDAP / UNIX groups that are allowed to login from
+#         any of the nodes in bastion_nodelist
+#         Note: can be empty
+# @example
+#   include profile::allow_ssh_from_login
+#
+class profile::ncsa::allow_ssh_from_login (
+    Array[ String ]    $allow_groups,
+    Array[ String, 1 ] $login_nodelist,
+    Hash               $custom_cfg,
+) {
+    $parms_local = {
+        'PubkeyAuthentication'   => 'yes',
+        'KerberosAuthentication' => 'yes',
+        'GSSAPIAuthentication'   => 'yes',
+    }
+
+    #add in allow_groups if non-empty
+    if size( $allow_groups ) > 0 {
+        $groups = { 'AllowGroups' => $allow_groups }
+    }
+    else {
+        $groups = {}
+    }
+    #join the param hashs
```

Comment:
General code review: when commenting code like this, comments should indicate why things are done, rather than what's being done. We can generally infer what the code is doing, so why are we doing the joining? Why is this behavior interesting and what should code readers know about what this means?

Reply:
Point taken. Removed redundant comment.